app/lib/json_web_token.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'uri'
+
+class JsonWebToken
+  AUTH0_DOMAIN = ENV["AUTH0_DOMAIN"]
+  JWT_JWKS_KEYS_URL = "https://#{AUTH0_DOMAIN}/.well-known/jwks.json"
+  JWT_TOKEN_ISS = "https://#{AUTH0_DOMAIN}/"
+
+  def self.decode_without_verification(token)
+    JWT.decode(token, nil, false)
+  end
+
+  def self.verify(token)
+    JWT.decode(
+      token, nil,
+      true, # Verify the signature of this token
+      algorithm: 'RS256',
+      iss: JWT_TOKEN_ISS,
+      verify_iss: true,
+      aud: Rails.application.secrets.auth0_api_audience,
+      verify_aud: true
+    ) do |header|
+      jwks_hash[header['kid']]
+    end
+  end
+
+  def self.jwks_hash
```

Comment:
could you add a comment or two to these methods about their function for future readers?

Reply:
Done